src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro256StarStarImpl.cs

```diff
@@ -90,46 +95,42 @@ public override int Next()
 
             public override int Next(int maxValue)
             {
-                if (maxValue > 1)
```

Comment:
Firstly intention is for it to never receive a negative maxValue, right? As the ImplBase this is protected in System.Random:
![image](https://user-images.githubusercontent.com/108188902/209328801-2d00526f-be69-4f65-a734-320b217dd75f.png)
Hence I assumed (perhaps wrongly) the check was merely there to avoid calculating the random in the trivial cases. I felt there was no need to make this check for all cases, just to save some calculation in the trivial cases.

What if it does receive a negative maxValue? Maybe (wrongly) used somewhere else internally... Since the algorithm actually continues with:
`uint usedMaxValue = (uint)maxValue`
in all places, and that it (in practice) will terminate for all uint values for this (imagined) usedMaxValue, I do not see a risk for a long nor an infinite loop for negative maxValue.

Currently it is not used anywhere else, as far as I can see.

Reply:
You're right, my mistake. They shouldn't be negative. I forgot that calls to XoshiroImpl go through the methods implemented in Random because the methods in XoshiroImpl override ImplBase and not Random. Maybe a `Debug.Assert` that the inputs really are valid would be appropriate?

Reply:
I think Debug.Asserts for non negative would be reasonable self-documentation.

Reply:
Added.